API/controllers/article-controller.js

```diff
@@ -0,0 +1,22 @@
+const { verifyID } = require("../models/meta-model");
+const { fetchArticle } = require("../models/article-model");
+
+exports.getArticle = (req, res, next) => {
+  const { article_id } = req.params;
+  return new Promise((resolve, reject) => {
+    if (!verifyID(article_id)) {
+      reject({ status: 400, msg: "ID is Malformed" });
+    } else resolve();
+  })
+    .then(() => {
+      return fetchArticle(article_id);
+    })
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Article not Found" });
+      }
+      const article = rows[0];
+      res.status(200).send({ article });
+    })
+    .catch(next);
```

Comment:
There are a few things here:
- Firstly this is a great handling of all the logic
- On a `400` where the ID is `NaN` , if the query is made with the invalid ID then `psql` will throw an error by default, which can then be handled by your error handling middleware checking for a code of `"22P02"`. So the meta-model logic is not required to handle this error
- This leads onto another question... if the model will reject a bad query, should it also reject from there if `rows.length === 0`. In order to keep effective separation of concerns I would move this logic over to the model, as it's already dealing with one error it should probably deal with them both